### ISSUE TITLE
applyPatch not using deep copy

### ISSUE THREAD

**Issue by callingmybluff** [chbrown/rfc6902#44]
I've looked around, and found #38 to be the most similar. However, I am facing a problem in the save category although that version is live probably.

# Bug of feature?
This is a bug, I believe.

# Use case / Example
https://jsfiddle.net/Ismaeel/1ewxshdq/

1. Use "Safe" to switch between the two versions of the text. It works as expected.
1. Use "Risky" to switch between the two versions of the text, and it fails. `versions[0]` takes something from `versions[1]`

# Expected vs. Current output
I would expect my patch not to be modified at all, but that is not the case for now.

**Comment by chbrown** [comment]
Good point. Fixed as of v3.0.2.

FWIW, your example was kinda confusing. The jsfiddle was unnecessary, and only distracted from your technically valid (and useful!) bug report. You could have just written something like:
```js
const value = {}
const numbers = []
applyPatch(value, [
  {op: 'add', path: '/numbers', value: numbers},
  {op: 'add', path: '/numbers/-', value: 9},
])
assert.ok(numbers.length == 0)
```
...which is pretty much how I wrote the [unit test](https://github.com/chbrown/rfc6902/blob/v3.0.2/test/issues.ts#L198-L209) for the bugfix.

**Comment by callingmybluff** [comment]
Wow!
I did make it a it complex I see 😶 

Thanks for the fix!
